### PR TITLE
cmake: use GNU linker on FreeBSD

### PR DESCRIPTION
**Pull request by wjwithagen**
The LLVM ld linker does not understand the versioning mapping.
This is a "documented" difference between GNU ld en LLVM ld.

So we fall back to the GNU linker in the binutils package.

Fixing this with a option to cmake is not possible when building
FreeBSD packages. Many of these options are filtered in the
ports make system.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by wjwithagen** [comment]
jenkins retest this please

File: do_freebsd.sh [hunk 1]
```diff
@@ -61,13 +58,12 @@ mkdir ${BUILD_DIR}
 	2>&1 | tee cmake.log
 
 echo -n "start building: "; date
-printenv
 
 cd ${BUILD_DIR}
-  gmake -j$CPUS V=1 VERBOSE=1 
+  gmake -j$NPROC V=1 VERBOSE=1
```
Comment (tchaikov) [inline do_freebsd.sh]: i am not sure how is this related to "Use the GNU binutils linker".

Comment (wjwithagen) [reply do_freebsd.sh]: @tchaikov 
That snuck in under the radar, since it is a typo/difference between what is actually set in the top of the file....
But I'll first try to see if your PR works for FreeBSD.

Major problem is that the FreeBSD ports infrastructure also sets quite a few CMake variables on calling `cmake`, so I really need to work around it, and doing it in the Ceph-config will be better resistant against future changes on the FreeSD side.

Comment (tchaikov) [reply do_freebsd.sh]: i think we discussed this before. my understanding is that, ceph tries to offer the machinery to work on different arch/distro/OS, and to avoid hardwiring any of them unless necessary. i think what you need is just make sure FreeBSD's ceph port use the machinery.

Comment (wjwithagen) [reply do_freebsd.sh]: @tchaikov 
Since  #30781 does not fix the FreeBSD problem.
Can I at least get this one in?

I've removed changes to do_freebsd.sh, so this is now a clean patch

Comment (tchaikov) [reply do_freebsd.sh]: could you change the title to something like

> cmake: use GNU linker on FreeBSD

?

Comment (wjwithagen) [reply do_freebsd.sh]: @tchaikov 
Ping?
